packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.tsx

```diff
@@ -86,11 +86,12 @@ const ButtonUnstyled = React.forwardRef(function ButtonUnstyled<
 
   const classes = useUtilityClasses(ownerState);
 
-  const Root: React.ElementType = component ?? components.Root ?? 'button';
+  const defaultElement = other.href || other.to ? 'a' : 'button';
+  const Root: React.ElementType = component ?? components.Root ?? defaultElement;
   const rootProps: WithOptionalOwnerState<ButtonUnstyledRootSlotProps> = useSlotProps({
     elementType: Root,
     getSlotProps: getRootProps,
-    externalForwardedProps: other,
+    externalForwardedProps: { ...other, href: other.href ?? other.to },
```

Comment:
Can you expand on why this is necessary?

Reply:
> Can you expand on why this is necessary?

Without this, the `href` attribute doesn't appear in the html.

Reply:
The `to` prop exists to make it easier to work with router links. It's not meant to be used as a replacement for the `href`. I'd pass through all the props as they are.

Reply:
> This got me thinking if the Button component is doing too much. We're basically encapsulating the behavior for links and buttons under the button component. Perhaps having a LinkButtonUnstyled (or LinkUnstyled) component would allow the implementation to be cleaner? Any thoughts, @mnajdova?

`Button` as link and `Link` component in my opinion are different components. On many web pages there are buttons that act as links, so supporting it in the `Button` component makes sense. Of course I am saying this from UI perspective, as those button looks like buttons, not links. 

The `Link` component in my opinion is an underlined text that contains a link, which is a different UI component. So I would say 👍 for adding link component, but 👎 for removing this logic from the `Button` :)